week7/answers.sql

```diff
@@ -1,3 +1,314 @@
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+DELIMITER ;;
+CREATE PROCEDURE equip(inventory_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT inventory_id INTO find_item_by_id;
+
+	SELECT 
+		inventory.character_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		inventory.item_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO equipped (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM inventory WHERE inventory.inventory_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE unequip(equipped_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT equipped_id INTO find_item_by_id;
+
+	SELECT 
+		equipped.character_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		equipped.item_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO inventory (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM equipped WHERE equipped.equipped_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE set_winners(team_id INT UNSIGNED)
+BEGIN
+	DECLARE find_winner_by_team_id INT UNSIGNED;
+	SELECT team_id INTO find_winner_by_team_id;
+
+	DELETE FROM winners;
+
+	INSERT INTO winners
+	SELECT 
+		characters.character_id,
+		characters.name
+	FROM teams
+		INNER JOIN team_members
+			ON teams.team_id = team_members.team_id
+		INNER JOIN characters
+			ON team_members.character_id = characters.character_id
+	WHERE teams.team_id = find_winner_by_team_id;
+END;;
+
+CREATE FUNCTION armor_total(character_id INT UNSIGNED)
+RETURNS TINYINT
+DETERMINISTIC 
+BEGIN
+	DECLARE find_armor_total_by_character_id INT UNSIGNED;
+	DECLARE armor_from_items TINYINT;
+	DECLARE natural_armor TINYINT;
+
+	SELECT character_id INTO find_armor_total_by_character_id;
+
+	SELECT 
+		SUM(items.armor)
+	FROM equipped
+		INNER JOIN items
+			ON equipped.item_id = items.item_id
+	WHERE equipped.character_id = find_armor_total_by_character_id
+	INTO armor_from_items;
+
+	SELECT character_stats.armor 
+	FROM character_stats 
+	WHERE character_stats.character_id = find_armor_total_by_character_id
+	INTO natural_armor;
+
+	Return armor_from_items + natural_armor;
+
+Return total_sum_of_armor;
+END;;
+
+CREATE PROCEDURE attack(id_of_character_being_attacked INT UNSIGNED, id_of_equipped_item_used_for_attack INT UNSIGNED)
+BEGIN
+	DECLARE find_dmg_by_item_id INT UNSIGNED;	
+	DECLARE character_being_attacked INT UNSIGNED;
+    
+	DECLARE weapon_dmg TINYINT;
+	DECLARE dmg_dealt TINYINT;
+	DECLARE character_armor TINYINT;
+	DECLARE character_hp TINYINT;
+
+	SELECT id_of_equipped_item_used_for_attack INTO find_dmg_by_item_id;
```

Comment:
Because of the pattern, I'm curious why argument variables are copied to locally-declared variables? You can save yourself these variable declarations and use the arguments directly.